internal/cow/cow.go

```diff
@@ -11,39 +12,39 @@ type Process interface {
 	// Close releases resources associated with the process and closes the
 	// writer and readers returned by Stdio. Depending on the implementation,
 	// this may also terminate the process.
-	Close() error
+	Close(ctx context.Context) error
 	// CloseStdin causes the process's stdin handle to receive EOF/EPIPE/whatever
 	// is appropriate to indicate that no more data is available.
-	CloseStdin() error
+	CloseStdin(ctx context.Context) error
 	// Pid returns the process ID.
 	Pid() int
 	// Stdio returns the stdio streams for a process. These may be nil if a stream
 	// was not requested during CreateProcess.
 	Stdio() (_ io.Writer, _ io.Reader, _ io.Reader)
 	// ResizeConsole resizes the virtual terminal associated with the process.
-	ResizeConsole(width, height uint16) error
+	ResizeConsole(ctx context.Context, width, height uint16) error
 	// Kill sends a SIGKILL or equivalent signal to the process and returns whether
 	// the signal was delivered. It does not wait for the process to terminate.
-	Kill() (bool, error)
+	Kill(ctx context.Context) (bool, error)
 	// Signal sends a signal to the process and returns whether the signal was
 	// delivered. The input is OS specific (either
 	// guestrequest.SignalProcessOptionsWCOW or
 	// guestrequest.SignalProcessOptionsLCOW). It does not wait for the process
 	// to terminate.
-	Signal(options interface{}) (bool, error)
+	Signal(ctx context.Context, options interface{}) (bool, error)
 	// Wait waits for the process to complete, or for a connection to the process to be
 	// terminated by some error condition (including calling Close).
 	Wait() error
 	// ExitCode returns the exit code of the process. Returns an error if the process is
 	// not running.
-	ExitCode() (int, error)
+	ExitCode(ctx context.Context) (int, error)
```

Comment:
It doesn't make sense to me at an interface level that ExitCode needs a context. We should be retrieving this internally when Wait resolves.

Reply:
Basically if the gcs package doesn't do anything with the context, we shouldn't include it here. We're going to stop using the internal bridge eventually, so I'd prefer not to muddy up the interface with confusing contexts in the meantime.

Reply:
I agree with you, however `cow.go` is more than External Bridge. `hcs/process.go` also implements `ExitCode` and in that case `ctx` is used because of the properties query to the guest. Should we favor external bridge?

Reply:
I guess it's fine for now since we still used the external bridge for WCOW. We can cut this once we use the external bridge everywhere.

Ideally hcs/process.go would change to get the exit code when the process wait is satisfied so that we have the context in the right place. But since we at least in theory support OpenProcess, we can't do this 100% of the time (same problem we had with Stdio and StdioLegacy).